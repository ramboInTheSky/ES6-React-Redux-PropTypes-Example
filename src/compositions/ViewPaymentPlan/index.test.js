import React from 'react';
import { shallow } from 'enzyme';

import { ViewPaymentPlanCompositions } from './';

describe('<ViewPaymentPlanCompositions />', () => {
  let props;
  let el;

  beforeEach(() => {
    props = {
      clearError: jest.fn(),
      currencySymbol: '£',
      error: false,
      errorText: {
        closePlan: 'There was a problem closing this payment plan, please try again.',
      },
      getPaymentPlan: jest.fn(),
      labels: {
        backButton: 'Back',
        closeButton: 'Close Payment Plan',
        installment: 'Installment',
        installmentAmount: 'Instalment amount',
        installmentFrequency: 'Instalment frequency',
        installmentPeriod: 'Instalment period',
        note: 'Note',
        paymentPlanType: 'Payment plan type',
        raisedBy: 'Raised by',
        startDate: 'Start date',
        submitButton: 'Create Payment Plan',
      },
      paymentPlan: {
        description:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod bibendum laoreet. Proin gravida dolor sit amet lacus accumsan et viverra justo commodo.',
        installment: '£40 every week from 23/07/2018',
        raisedBy: 'Mabelle Ortega',
        endDate: '23/12/2018',
        startDate: '23/07/2018',
        type: 'Personal',
      },
      onBack: jest.fn(),
      onClose: jest.fn(),
      updatePageHeader: jest.fn(),
    };

    el = shallow(<ViewPaymentPlanCompositions {...props} />);
  });

  it('should render the page', () => {
    expect(el).toMatchSnapshot();
  });

  it('should render the page with an error message', () => {
    el = shallow(<ViewPaymentPlanCompositions {...props} error />);
    expect(el).toMatchSnapshot();
  });

  it('should call updatePageHeader', () => {
    expect(props.updatePageHeader).toHaveBeenCalled();
  });

  it('should call getPaymentPlan', () => {
    expect(props.getPaymentPlan).toHaveBeenCalled();
  });

  it('should call clearError on unmount', () => {
    el.unmount();
    expect(props.clearError).toHaveBeenCalled();
  });
});
