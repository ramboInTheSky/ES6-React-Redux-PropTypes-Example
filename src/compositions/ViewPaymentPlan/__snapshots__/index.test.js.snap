// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewPaymentPlanCompositions /> should render the page 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <styled.div>
      <Styled(styled.h3)>
        View payment plan
      </Styled(styled.h3)>
      <styled.section>
        <Styled(styled.h4)>
          Payment plan type
        </Styled(styled.h4)>
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-type"
        >
          Personal
        </Styled(styled.p)>
      </styled.section>
      <styled.section>
        <Styled(styled.h4)>
          Installment
        </Styled(styled.h4)>
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-installment"
        >
          £40 every week from 23/07/2018
        </Styled(styled.p)>
      </styled.section>
      <styled.section>
        <Styled(styled.h4)>
          Start date
        </Styled(styled.h4)>
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-startDate"
        >
          23/07/2018
        </Styled(styled.p)>
      </styled.section>
      <styled.section>
        <Styled(styled.h4) />
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-endDate"
        >
          23/12/2018
        </Styled(styled.p)>
      </styled.section>
      <styled.section>
        <Styled(styled.h4)>
          Raised by
        </Styled(styled.h4)>
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-raisedBy"
        >
          Mabelle Ortega
        </Styled(styled.p)>
      </styled.section>
      <styled.section>
        <Styled(styled.h4)>
          Note
        </Styled(styled.h4)>
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-description"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod bibendum laoreet. Proin gravida dolor sit amet lacus accumsan et viverra justo commodo.
        </Styled(styled.p)>
      </styled.section>
      <ActionButtonsWrapper
        message=""
      >
        <WithTheme(Button)
          buttonType="secondary"
          data-bdd="viewPaymentPlan-back"
          isFullWidth={true}
          onClick={[Function]}
        >
          Back
        </WithTheme(Button)>
        <WithTheme(Button)
          data-bdd="viewPaymentPlan-close"
          isFullWidth={true}
          onClick={[Function]}
        >
          Close Payment Plan
        </WithTheme(Button)>
      </ActionButtonsWrapper>
    </styled.div>
  </div>
</withRouter(Connect(PageContentComposition))>
`;

exports[`<ViewPaymentPlanCompositions /> should render the page with an error message 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <styled.div>
      <styled.div
        data-bdd="viewPaymentPlan-error"
      >
        <NotificationPanel
          description="There was a problem closing this payment plan, please try again."
          hideCloseButton={true}
          icon="warning"
          onCloseNotification={[Function]}
          show={true}
        />
      </styled.div>
      <Styled(styled.h3)>
        View payment plan
      </Styled(styled.h3)>
      <styled.section>
        <Styled(styled.h4)>
          Payment plan type
        </Styled(styled.h4)>
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-type"
        >
          Personal
        </Styled(styled.p)>
      </styled.section>
      <styled.section>
        <Styled(styled.h4)>
          Installment
        </Styled(styled.h4)>
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-installment"
        >
          £40 every week from 23/07/2018
        </Styled(styled.p)>
      </styled.section>
      <styled.section>
        <Styled(styled.h4)>
          Start date
        </Styled(styled.h4)>
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-startDate"
        >
          23/07/2018
        </Styled(styled.p)>
      </styled.section>
      <styled.section>
        <Styled(styled.h4) />
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-endDate"
        >
          23/12/2018
        </Styled(styled.p)>
      </styled.section>
      <styled.section>
        <Styled(styled.h4)>
          Raised by
        </Styled(styled.h4)>
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-raisedBy"
        >
          Mabelle Ortega
        </Styled(styled.p)>
      </styled.section>
      <styled.section>
        <Styled(styled.h4)>
          Note
        </Styled(styled.h4)>
        <Styled(styled.p)
          data-bdd="viewPaymentPlan-description"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod bibendum laoreet. Proin gravida dolor sit amet lacus accumsan et viverra justo commodo.
        </Styled(styled.p)>
      </styled.section>
      <ActionButtonsWrapper
        message=""
      >
        <WithTheme(Button)
          buttonType="secondary"
          data-bdd="viewPaymentPlan-back"
          isFullWidth={true}
          onClick={[Function]}
        >
          Back
        </WithTheme(Button)>
        <WithTheme(Button)
          data-bdd="viewPaymentPlan-close"
          isFullWidth={true}
          onClick={[Function]}
        >
          Close Payment Plan
        </WithTheme(Button)>
      </ActionButtonsWrapper>
    </styled.div>
  </div>
</withRouter(Connect(PageContentComposition))>
`;
